lesson14_homework_interface/homework_14.py

```diff
@@ -0,0 +1,111 @@
+import json
+
+
+class BaseHandler:
+    def __init__(self, path):
+        self.__path = path
+        self.content = None
+        self._file = None
+
+    @property
+    def path(self):
+        return self.__path
+
+    def read(self):
+        raise NotImplementedError
+
+    def append(self, new_string):
+        raise NotImplementedError
+
+    def close(self):
+        raise NotImplementedError
+
+
+class JsonHandler(BaseHandler):
+    def __init__(self, path):
+        super().__init__(path)
+        try:
+            with open(self.path, 'r') as self._file:
+                self.content = json.load(self._file)
+            quit('this json file format is not supported') if not isinstance(self.content, list) else ...
+        except FileNotFoundError:
+            pass
+
+    def read(self):
+        try:
+            with open(self.path, 'r') as self._file:
+                self.content = json.load(self._file)
+        except FileNotFoundError:
+            with open(self.path, 'w') as self._file:
+                json.dump([], self._file)
+            with open(self.path, 'r') as self._file:
+                self.content = json.load(self._file)
+        return self.content
+
+    def append(self, new_string):
+        self.content.append(new_string)
+
+    def close(self):
+        with open(self.path, 'w') as self._file:
+            json.dump(self.content, self._file)
+
+
+class TxtHandler(BaseHandler):
+    def read(self):
+        try:
+            with open(self.path, 'r') as self._file:
+                self.content = self._file.read()
+        except FileNotFoundError:
```

Comment:
and here..

Reply:
I mean that this is unnecessary

Reply:
+ why you read method check existing file? move it to another place if you want..
You can do it into Base class or define method for it into Base class